activestorage/test/models/attachments_test.rb

```diff
@@ -146,6 +146,18 @@ class ActiveStorage::AttachmentsTest < ActiveSupport::TestCase
     assert_predicate @user.avatar, :identified?
   end
 
+  test "identify newly-attached blob before analyzing" do
+    perform_enqueued_jobs do
+      blob = create_file_blob
+      blob.update!(metadata: {})
+      @user.avatar.attach blob
+    end
```

Comment:
The situation in which we actually take this code path is attaching a directly-uploaded blob, so I‘d prefer to simulate a direct upload like we do in the test above (`test "identify newly-attached, directly-uploaded blob"`). This is a good opportunity to extract a new helper method with an intention-revealing name for both tests to use.